### PR TITLE
Simple overload for StartTranslation

### LINKED ISSUE #19494: Initialize input collections so user can do `.Add()`
Feedback from Archboard meeting. #19348

### PR DESCRIPTION
**Pull request by a-noyass**
Fixes: #19494 

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running generate.ps1. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version,

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: sdk/documenttranslation/Azure.AI.DocumentTranslation/src/TranslationConfiguration.cs [hunk 1]
```diff
@@ -9,5 +10,54 @@ namespace Azure.AI.DocumentTranslation
...
+        /// <summary>
+        /// Initializes a new instance of TranslationConfiguration.
+        /// </summary>
+        /// <param name="sourceUri">The SAS URL for the source container containing documents to be translated.</param>
+        /// <param name="targetUri">The SAS URL for the source container containing documents to be translated.</param>
+        /// <param name="targetLanguageCode">Language code to translate documents to. For supported languages see
+        /// <a href="https://docs.microsoft.com/azure/cognitive-services/translator/language-support#translate"/>.</param>
+        /// <param name="glossary">Custom translation glossary to be used in the translation operation. For supported file types see
+        /// <see cref="DocumentTranslationClient.GetGlossaryFormatsAsync(System.Threading.CancellationToken)"/>.</param>
+        public TranslationConfiguration(Uri sourceUri, Uri targetUri, string targetLanguageCode, TranslationGlossary glossary = default)
```
Comment (a-noyass) [inline sdk/documenttranslation/Azure.AI.DocumentTranslation/src/TranslationConfiguration.cs:22]: @maririos should these be named sourceBlobContainerSas and targetBlobContainerSas like we had in the old overload?

Comment (maririos) [reply sdk/documenttranslation/Azure.AI.DocumentTranslation/src/TranslationConfiguration.cs:22]: hmmm i think for now let's leave them generic. From the arch board, other approach is to add a `FromAzureStorage` method where we could use those names, but I don't think it is necessary for now

File: sdk/documenttranslation/Azure.AI.DocumentTranslation/src/DocumentTranslationClient.cs [hunk 2]
```diff
@@ -155,49 +155,18 @@ public virtual async Task<DocumentTranslationOperation> StartTranslationAsync(IE
...
         /// </summary>
-        /// <param name="sourceBlobContainerSas">The SAS URL for the source container containing documents to be translated. </param>
-        /// <param name="targetBlobContainerSas">The SAS URL for the target container to which the translated documents will be written. </param>
-        /// <param name="targetLanguageCode">Language code to translate documents to. For supported languages see
-        /// <a href="https://docs.microsoft.com/azure/cognitive-services/translator/language-support#translate"/>.</param>
-        /// <param name="glossary">Custom translation glossary to be used in the translation operation. For supported file types see
-        /// <see cref="GetGlossaryFormatsAsync(CancellationToken)"/>.</param>
-        /// <param name="options">Set translation options including source language and custom translation category.</param>
+        /// <param name="configuration">Sets the configurations for the translation operation
+        /// including source and target storage for documents to be translated. </param>
```
Comment (maririos) [inline sdk/documenttranslation/Azure.AI.DocumentTranslation/src/DocumentTranslationClient.cs:163]:
```suggestion
        /// including source and target containers for documents to be translated. </param>
```

File: sdk/documenttranslation/Azure.AI.DocumentTranslation/src/DocumentTranslationClient.cs [hunk 3]
```diff
@@ -214,50 +183,18 @@ public virtual DocumentTranslationOperation StartTranslation(Uri sourceBlobConta
...
```
Comment (maririos) [inline sdk/documenttranslation/Azure.AI.DocumentTranslation/src/DocumentTranslationClient.cs:191]:
```suggestion
        /// including source and target containers for documents to be translated. </param>
```

File: sdk/documenttranslation/Azure.AI.DocumentTranslation/src/TranslationConfiguration.cs [hunk 4]
```diff
@@ -9,5 +10,54 @@ namespace Azure.AI.DocumentTranslation
     [CodeGenModel("BatchRequest")]
     public partial class TranslationConfiguration
     {
+        /// <summary>
+        /// Initializes a new instance of TranslationConfiguration.
+        /// </summary>
+        /// <param name="sourceUri">The SAS URL for the source container containing documents to be translated.</param>
+        /// <param name="targetUri">The SAS URL for the source container containing documents to be translated.</param>
```
Comment (maririos) [inline sdk/documenttranslation/Azure.AI.DocumentTranslation/src/TranslationConfiguration.cs:17]: needs update

File: sdk/documenttranslation/Azure.AI.DocumentTranslation/src/TranslationConfiguration.cs [hunk 5]
```diff
@@ -9,5 +10,54 @@ namespace Azure.AI.DocumentTranslation
...
+        /// <param name="targetLanguageCode">Language code to translate documents to. For supported languages see
+        /// <a href="https://docs.microsoft.com/azure/cognitive-services/translator/language-support#translate"/>.</param>
+        /// <param name="glossary">Custom translation glossary to be used in the translation operation. For supported file types see
```
Comment (maririos) [inline sdk/documenttranslation/Azure.AI.DocumentTranslation/src/TranslationConfiguration.cs:20]: should we do something like `Custom <see cref="TranslationGlossary"/> to be used...`

File: sdk/documenttranslation/Azure.AI.DocumentTranslation/src/TranslationConfiguration.cs [hunk 6]
```diff
@@ -9,5 +10,54 @@ namespace Azure.AI.DocumentTranslation
...
+            {
+                target.Glossaries.Add(glossary);
+            }
+            Targets.Add(target);
+        }
+
+        /// <summary> Initializes a new instance of TranslationConfiguration. </summary>
+        /// <param name="source"> Source of the input documents. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="source"/> is null. </exception>
+        public TranslationConfiguration(TranslationSource source)
```
Comment (maririos) [inline sdk/documenttranslation/Azure.AI.DocumentTranslation/src/TranslationConfiguration.cs:36]: consider deleting this method

File: sdk/documenttranslation/Azure.AI.DocumentTranslation/src/TranslationConfiguration.cs [hunk 7]
```diff
@@ -9,5 +10,54 @@ namespace Azure.AI.DocumentTranslation
...
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+            Source = source;
+        }
+
+        /// <summary>
+        /// Add Translation Target to the configuration.
+        /// </summary>
+        /// <param name="targetUri">The SAS URL for the source container containing documents to be translated.</param>
```
Comment (maririos) [inline sdk/documenttranslation/Azure.AI.DocumentTranslation/src/TranslationConfiguration.cs:48]: needs update

File: sdk/documenttranslation/Azure.AI.DocumentTranslation/src/TranslationConfiguration.cs [hunk 8]
```diff
@@ -9,5 +10,54 @@ namespace Azure.AI.DocumentTranslation
...
+        /// <param name="languageCode">Language code to translate documents to. For supported languages see
+        /// <a href="https://docs.microsoft.com/azure/cognitive-services/translator/language-support#translate"/>.</param>
+        /// <param name="glossary">Custom translation glossary to be used in the translation operation. For supported file types see
```
Comment (maririos) [inline sdk/documenttranslation/Azure.AI.DocumentTranslation/src/TranslationConfiguration.cs:51]: same comment about adding cref

**Review by maririos — APPROVED** [review]

**Pull request merged**